nbdime/gitmergetool.py

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env python
+"""A git mergetool plugin for notebooks.
+
+Uses nbdime to merge notebooks instead of plain text merges of JSON.
+
+Enable in your global git config with:
+
+    git-nbmergetool config --enable [--global]
+
+Use with:
+
+    git mergetool [<commit> [<commit>]]
+"""
+import os
+import sys
+from subprocess import check_call, check_output, CalledProcessError
+
+from . import nbmergeapp
+
+def enable(global_=False):
+    """Enable nbdime git mergetool"""
+    cmd = ['git', 'config']
+    if global_:
+        cmd.append('--global')
+
+    # Register CLI tool
+    check_call(cmd + ['mergetool.nbdime.cmd', 'git-nbmergetool merge "$LOCAL" "$REMOTE" "$BASE" "$MERGED"'])
+    try:
+        previous = check_output(cmd + ['merge.tool']).decode('utf8', 'replace').strip()
+    except CalledProcessError:
+        previous = None
+    else:
+        if previous != 'nbdime':
+            check_call(cmd + ['mergetool.nbdime.previous', previous])
+
+    # Register webapp tool
+    check_call(cmd + ['mergetool.nbdimeweb.cmd', 'git-nbwebmergetool "$LOCAL" "$REMOTE" "$BASE" "$MERGED"'])
+
+    # Common setting:
+    check_call(cmd + ['mergetool.prompt', 'false'])
+
+    # Set default tool to webapp
+    check_call(cmd + ['merge.tool', 'nbdimeweb'])
+
+
+def disable(global_=False):
+    """Disable nbdime git mergetool"""
+    cmd = ['git', 'config']
+    if global_:
+        cmd.append('--global')
+    try:
+        previous = check_output(cmd + ['mergetool.nbdime.previous']).decode('utf8', 'replace').strip()
+    except CalledProcessError:
+        try:
+            check_call(cmd + ['--unset', 'merge.tool'])
+        except CalledProcessError:
+            # already unset
+            pass
+    else:
+        check_call(cmd + ['merge.tool', previous])
+
+
+def show_merge(base, local, remote, merged):
+    """Run the mergetool
+
+    If we are merging a notebook, show the merge via nbmerge.
+    Otherwise, exit with error code.
+    """
+    # TODO: handle /dev/null (Windows equivalent?) for new or deleted files
+    if any([not f.endswith('.ipynb') for f in [base, local, remote, merged]]):
```

Comment:
Tip: `any([x for x in mylist])` will build the list first and then run `any` on it. Instead, use a generator by dropping `[]`, i.e.:  `any(x for x in mylist)`  to let `any` abort on first True value. Not important here, but slightly less typing and improves performance.